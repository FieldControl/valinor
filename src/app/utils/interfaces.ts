export interface Indexable {
  [key: string]: any;
}

export interface User {
  name: string;
  email: string;
  login?: string;
  id: number;
  node_id: string;
  avatar_url: string;
  gravatar_id: string;
  url: string;
  html_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  starred_url: string;
  subscriptions_url: string;
  organizations_url: string;
  repos_url: string;
  events_url: string;
  received_events_url: string;
  type: string;
  site_admin: boolean;
  starred_at: string;
}

export interface License {
  key: string;
  name: string;
  url: string;
  spdx_id: string;
  node_id: string;
  html_url: string;
}

export interface SearchResultTextMatch {
  object_url: string;
  object_type: string;
  property: string;
  fragment: string;
  matches: { text: string; indices: number[] }[];
}

export interface RepoSearchResultItem {
  id: number;
  node_id: string;
  name?: string;
  full_name?: string;
  owner?: User;
  private: boolean;
  html_url: string;
  description?: string;
  fork: boolean;
  url: string;
  created_at: string;
  updated_at: string;
  pushed_at?: string;
  homepage: string;
  size: number;
  stargazers_count?: number;
  watchers_count: number;
  language?: string;
  forks_count: number;
  open_issues_count: number;
  master_branch: string;
  default_branch: string;
  score: number;
  forks_url: string;
  keys_url: string;
  collaborators_url: string;
  teams_url: string;
  hooks_url: string;
  issue_events_url: string;
  events_url: string;
  assignees_url: string;
  branches_url: string;
  tags_url: string;
  blobs_url: string;
  git_tags_url: string;
  git_refs_url: string;
  trees_url: string;
  statuses_url: string;
  languages_url: string;
  stargazers_url: string;
  contributors_url: string;
  subscribers_url: string;
  subscription_url: string;
  commits_url: string;
  git_commits_url: string;
  comments_url: string;
  issue_comment_url: string;
  contents_url: string;
  compare_url: string;
  merges_url: string;
  archive_url: string;
  downloads_url: string;
  issues_url: string;
  pulls_url: string;
  milestones_url: string;
  notifications_url: string;
  labels_url: string;
  releases_url: string;
  deployments_url: string;
  git_url: string;
  ssh_url: string;
  clone_url: string;
  svn_url: string;
  forks: number;
  open_issues: number;
  watchers: number;
  topics?: string[];
  mirror_url: string;
  has_issues: boolean;
  has_projects: boolean;
  has_pages: boolean;
  has_wiki: boolean;
  has_downloads: boolean;
  has_discussions: boolean;
  archived: boolean;
  disabled: boolean;
  visibility: string;
  license: License;
  permissions: {
    admin: boolean;
    maintain: boolean;
    push: boolean;
    triage: boolean;
    pull: boolean;
  };
  text_matches: SearchResultTextMatch[];
  temp_clone_token: string;
  allow_merge_commit: boolean;
  allow_squash_merge: boolean;
  allow_rebase_merge: boolean;
  allow_auto_merge: boolean;
  delete_branch_on_merge: boolean;
  allow_forking: boolean;
  is_template: boolean;
  web_commit_signoff_required: boolean;
}

export interface RepoSearchResponse {
  total_count: number;
  incomplete_results: boolean;
  items: RepoSearchResultItem[];
}
