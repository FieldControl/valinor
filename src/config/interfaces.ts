interface IRepositoryOwner {
  login: string;
}

export interface IRepository {
  id: number;
  name: string;
  owner: IRepositoryOwner;
  html_url: string;
  language: string;
  updated_at: string;
  description: string;
  stargazers_count: number;
  open_issues_count: number;
}
