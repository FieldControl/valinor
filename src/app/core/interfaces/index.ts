export * from './system-route.interface';
export * from './auth.interface';
