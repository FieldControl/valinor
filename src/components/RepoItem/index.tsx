import React from "react";
import TagItem from "../TagItem";

import {
  Container,
  BookMarkIcon,
  TextContent,
  Header,
  Name,
  Description,
  TagList,
  Footer,
  StarsContainer,
  Star,
  Language,
  Updated,
  Issues,
} from "./styles";

const RepoItem: React.FC = () => {
  return (
    <Container>
      <BookMarkIcon />

      <TextContent>
        <Header>
          <Name href="#">nodejs/node</Name>
          <Description>Node.js JavaScript runtime</Description>
        </Header>

        <TagList>
          <TagItem />
          <TagItem />
          <TagItem />
          <TagItem />
          <TagItem />
          <TagItem />
        </TagList>

        <Footer>
          <StarsContainer>
            <Star />
            81.7k
          </StarsContainer>
          <Language>JavaScript</Language>
          <Updated>Updated 3 hours ago</Updated>
          <Issues href="#">75 issues need help</Issues>
        </Footer>
      </TextContent>
    </Container>
  );
};

export default RepoItem;
