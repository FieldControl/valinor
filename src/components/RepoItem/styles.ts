import styled, { css } from "styled-components";
import { RiBookMarkLine, RiStarLine } from "react-icons/ri";

const iconCss = css`
  width: 16px;
  height: 16px;
  fill: #6a737d;
  flex-shrink: 0;
`;

export const Star = styled(RiStarLine)`
  ${iconCss}
`;

export const BookMarkIcon = styled(RiBookMarkLine)`
  ${iconCss}
  height: 24px;
`;

export const Container = styled.div`
  border-top: 1px solid #d0d7de;
  display: flex;
  padding: 24px 0;
`;

export const TextContent = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-start;
  flex-direction: column;
  margin-left: 8px;
`;

export const Header = styled.div``;

export const Name = styled.a`
  text-decoration: none;
  color: #0869da;
  line-height: 24px;

  &:hover {
    text-decoration: underline;
  }
`;

export const Description = styled.p`
  color: #24292f;
  margin-bottom: 4px;
  line-height: 21px;
`;

export const Footer = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
`;

export const StarsContainer = styled.button`
  color: #57606a;
  border: none;
  cursor: pointer;
  font-size: 12px;
  background-color: transparent;
  margin-right: 16px;
  display: flex;
  align-items: center;

  ${Star} {
    margin-right: 4px;
  }

  &:hover {
    color: #58a6ff;

    ${Star} {
      fill: #58a6ff;
    }
  }
`;

export const Language = styled.p`
  color: #57606a;
  font-size: 12px;
  margin-right: 16px;
`;

export const Updated = styled.p`
  color: #57606a;
  font-size: 12px;
  margin-right: 16px;
`;

export const Issues = styled.a`
  color: #57606a;
  text-decoration: none;
  font-size: 12px;

  &:hover {
    color: #58a6ff;
  }
`;
