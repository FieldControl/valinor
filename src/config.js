// URL de conex√£o do banco
module.exports = {
  db: {
    // database: 'mongodb://admin:voafield10@ds221435.mlab.com:21435/backend-challenge-production'
    database: 'mongodb://test:testfieldcontrol10@ds123675.mlab.com:23675/backend-challenge-test'
  }
}
