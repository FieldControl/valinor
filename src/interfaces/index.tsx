export interface fileDetails {
  name: string,
  type: string,
}

export interface imagePath {
  [key: string]: string,
}

export interface repoData {
  id: number,
  name: string,
  full_name: string,
  description: string,
  topics: string[],
  updated_at: string,
  language: string,
}

export interface headerProps {
  barVisibily: boolean,
}

export interface cardProps {
  data: {
    id: number,
    name: string,
    full_name: string,
    description: string,
    topics: string[],
    updated_at: string,
    language: string,
  }
}

export interface readmeProps {
  user: string | undefined,
  repo: string | undefined,
}
