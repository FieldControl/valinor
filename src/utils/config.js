
const isProduction = (process.env.PROD || 'false') == 'true'

const appUrl = isProduction ? 'https://gql-leo.herokuapp.com/' : 'http://localhost:4000'

module.exports = {
  isProduction,
  appUrl,
  databaseCredentials: {
    ...credentialsFromEnv()
  }
}

function credentialsFromEnv () {
  // exemple url mysql://user:pass@host:port/database?reconnect=true
  const connectionString = process.env.GQL_DATABASE_URL
  // mysql://tsv0j2hl5m7q14l2:y1uzfzvtvo7kuwhp@g3v9lgqa8h5nq05o.cbetxkdyhwsb.us-east-1.rds.amazonaws.com:3306/c572ymepzv3e1tq5
  // const connectionString = 'mysql://root:202cb962ac59075b964b07152d234b70@localhost:3306/gqltest?reconnect=true'

  console.log(connectionString)
  const url = 'mysql://' + connectionString.split('@')[1]
  const user = connectionString.split('://')[1].split(':')[0]
  const password = connectionString.split('://')[1].split(':')[1].split('@')[0]
  const host = connectionString.split('@')[1].split(':')[0]
  const database = connectionString.split('@')[1].split('/')[1].split('?')[0]
  const port = connectionString.split('@')[1].split(':')[1].split('/')[0]
  return {
    host,
    url,
    user,
    password,
    database,
    port
  }
}
