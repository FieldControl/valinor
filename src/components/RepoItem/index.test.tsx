import { shallow } from "enzyme";
import RepoItem from ".";
import { IRepository } from "../../config/interfaces";

describe("RepoItem", () => {
  const REPOSITORY: IRepository = {
    id: 123,
    name: "Test repository",
    owner: { login: "userowner" },
    html_url: "https://test.repository",
    language: "Javascript",
    updated_at: "1631743676394",
    description: "A repository mock",
    stargazers_count: 2000,
    open_issues_count: 20,
  };

  it("should match the snapshot", () => {
    const wrapper = shallow(<RepoItem repository={REPOSITORY} />);
    expect(wrapper).toMatchSnapshot();
  });
});
