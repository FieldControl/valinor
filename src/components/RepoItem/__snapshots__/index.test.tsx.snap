// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoItem should match the snapshot 1`] = `
<styled.div>
  <Styled(RiBookMarkLine) />
  <styled.div>
    <styled.div>
      <styled.a
        href="https://test.repository"
      >
        userowner
        /
        <strong>
          Test repository
        </strong>
      </styled.a>
      <styled.p>
        A repository mock
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.button>
        <Styled(RiStarLine) />
        2.00K
      </styled.button>
      <styled.p>
        Javascript
      </styled.p>
      <styled.p>
        Updated 
        385 years ago
      </styled.p>
      <styled.a
        href="https://test.repository/issues"
      >
        20
         issues need help
      </styled.a>
    </styled.div>
  </styled.div>
</styled.div>
`;
